Standardise component structure (#150)
sa-tre/satre-specification: docs/source/pillars/data_management.md

  - Decisions that are made as part of the process discussed above should be recorded and made available for inspection by all stakeholders.
  - Recommended
* - You must have a data ingress process which enforces information governance rules/processes.
  - The data ingress process needs to ensure that information governance is correctly followed.
    In particular, it should require that an ingress request has been approved by all required parties.
  - Mandatory
* - You must have a data egress process which enforces information governance rules/processes.
  - The data egress process needs to ensure that information governance requirements are adhered to.
    In particular, it should require that an egress request has been approved by all required parties.
  - Mandatory
* - Your data egress process could sometimes require project-independent approval.
  - There may be cases where there are multiple stakeholders for a piece of analysis including data providers, data analysts, data subjects, the TRE operator.
    A data egress process may then require approval from people not on the project team, for example an external referee or TRE operator representative
  - Optional

> * - You must keep a record of what data it holds.

Reviewer: ```suggestion
* - You must keep a record of what data your TRE holds.
```

  - Good records are important for ensuring compliance with legislation, understanding risk and aiding good data hygiene.
    The record should include a description of the data, its source, contact details for the data owner, which projects use the data, the date it was received, when it is expected to no longer be needed.
  - Mandatory
* - You must have a policy on data deletion.
  - There should be a clear, published policy on when data will be retained or deleted.
    This may allow time for data owners to consider outputs they may want to extract from the TRE.
    Any sensitive data, including all backups, should be deleted when they are no longer needed.
    Having clear policies will help to avoid problems with data being kept longer than necessary or accidental deletion of outputs.
  - Mandatory
* - You could keep backups of data and research environments, provided that this is permitted by law.
  - Keeping backups could help reduce the impact of events like accidental deletion and data corruption on work in a TRE.

>     TRE developers may want to consider how different elements, for example sensitive input data or users workspaces, may be backed up or if they should be.

Reviewer: ```suggestion
    TRE developers may want to consider how different elements such as sensitive input data or users' workspaces may be backed up, and whether they should be.
```

* - Statement
  - Guidance
  - Importance
* - You must not create user accounts for use by more than one person.
  - It is important that each user account should be used by one, and only one, person in order to facilitate the assignment of roles or permissions and to log the actions of individuals.
  - Mandatory
* - You must be reasonably convinced of the identity of each person being granted an account.

>   - It is important to ensure access, via an account, has been given to the correct person.

Reviewer: ```suggestion
  - It is important to ensure an account has been given to the correct person.
```

    For example, multiple credentials may be used before account creation to verify identity or, when appropriate, photo ID checks may be required.
  - Mandatory
* - You must restrict a user's access to only data required in their work.
  - There is no need to grant an individual access to data they do not require.
    Access may be assigned in a manner appropriate to a TREs design, for example through roles granted to user accounts or through isolated project workspaces.
  - Mandatory
* - You must ensure that multi-factor authentication is enabled for all users.
  - Multi-factor authentication ensures that to successfully connect a user must have more than one piece of evidence in different categories.
    Categories include something the user knows (_e.g._ a password), something the user possesses (_e.g._ a TOTP key) or something the user is (_e.g._ biometric data).
    A TRE does not need to implement multi-factor authentication checks itself if it is provided by a third-party identity provider.
  - Mandatory

> * - You could use federated authentication/ single sign-on (SSO) for user login.

Reviewer: ```suggestion
* - You could use federated authentication or single sign-on (SSO) for user login.
```

* - Statement
  - Guidance
  - Importance
* - You should carry out penetration tests on your TRE.
  - By intentionally attempting to breach their TRE, organisations can proactively discover unnoticed vulnerabilities before they are exploited maliciously. Tests can evaluate the effectiveness of security controls in preventing data breaches, unauthorised access, or other security incidents.
  - Recommended
* - You should update the security controls of your TRE based on the results of security tests.
  - Security testing can reveal bugs and discrepancies in the TRE architecture which should be addressed in advance of sensitive data being uploaded, or with urgency in the case of an operational TRE.
    Regular testing will allow organisations to refine their TRE security controls and incident response capabilities, enabling them to adapt to any new security concerns that may arise as a result of changes in the underlying software.
  - Recommended
* - You must have procedures in place for rapid incident response.

>   - There may well be legal requirements to disclose details of any incidents, _e.g._ data breaches for organisations subject to GDPR.

Reviewer: ```suggestion
  - There may be legal requirements to disclose details of any incidents, such as  data breaches for organisations subject to GDPR.
```

* - Statement
  - Guidance
  - Importance
* - Your TRE must encrypt project and user data at rest.
  - This prevents unauthorised access to the data even if the storage media is compromised.
    This may involve encrypted filesystems or tools to encrypt and decrypt data on demand.
    The encryption keys may be managed by the TRE operator or by a trusted external actor (_e.g._ a cloud services provider).
  - Mandatory

> * - Your TRE must encrypt data when in transit between the TRE and the outside world.

Author: ```suggestion
* - Your TRE must encrypt data when in transit between the TRE and external networks or computers.
```
